app/models/veteran_onboarding.rb

```diff
@@ -17,16 +17,36 @@ class VeteranOnboarding < ApplicationRecord
   belongs_to :user_account, primary_key: :id, foreign_key: :user_account_uuid, inverse_of: :veteran_onboarding
   validates :user_account, uniqueness: true
 
+  attr_accessor :user
+
   # Determines whether the onboarding flow should be displayed for a veteran.
-  # Currently, this is based solely on the value of the `display_onboarding_flow` attribute.
-  # In the future, additional information (like MPI info) might be taken into account.
+  # Currently, we have two feature toggle checks:
+  # - veteran_onboarding_show_to_newly_onboarded examines settings for veteran_onboarding.onboarding_threshold_days,
+  #   (default 180 days) and examines the user_verification.verified_at attribute
+  # - veteran_onboarding_beta_flow just checks if the feature toggle is enabled for the user
   def show_onboarding_flow_on_login
-    display_onboarding_flow
+    if @user.user_account&.verified?
+      if Flipper.enabled?(:veteran_onboarding_show_to_newly_onboarded, @user)
+        days_since_verification = (Time.zone.today - @user.user_verification.verified_at.to_date).to_i
```

Comment:
Since a `UserAccount` can have multiple `UserVerifications`, is it intended that this would trigger and show onboarding flows for users that have switched from an established `UserVerification` that is past the display time to a new one?

Reply:
Ah that's a good point.  I can see that the user can have multiple UserVerifications:

```
  def get_user_verification
    case identity_sign_in&.dig(:service_name)
    when SAML::User::MHV_ORIGINAL_CSID
      return UserVerification.find_by(mhv_uuid: mhv_correlation_id) if mhv_correlation_id
    when SAML::User::DSLOGON_CSID
      return UserVerification.find_by(dslogon_uuid: identity.edipi) if identity.edipi
    when SAML::User::LOGINGOV_CSID
      return UserVerification.find_by(logingov_uuid:) if logingov_uuid
    end
    return nil unless idme_uuid

    UserVerification.find_by(idme_uuid:) || UserVerification.find_by(backing_idme_uuid: idme_uuid)
  end
```

I assumed that it wasn't actually likely that they would verify their identity multiple times though.  Would that actually happen?

In such a case, if a user had verified with mhv, for example, that one is going to be always returned, even if they later verify with idme.  Because of the order that we retrieve the UserVerification record using that case statement.  The only circumstance where the onboarding window could be "reset" would be if they first verified with ex: idme, and then by login dot gov, and then by ds login, or whatever.

I sortof feel like this is an edge case which is maybe not that critical to cover?  I am happy to address it if you think it's important.  

One way that I could address this is by adding a bit of logic that would update the `display_onboarding_flow` value to `false` in the database if the user's `verified_at` record is past the threshold in days.  That would prevent the clock from being reset to 0 if they verified again.  That said, it wouldn't prevent the resetting of the clock if the user re-verified within that time threshold.

Reply:
I should clarify: it wouldn't reset the clock, but it *would* disable the onboarding flow from automatically being displayed to the user when they log in

Reply:
The `user.user_verification` will always be whichever credential the user has currently authenticated with, the `identity_sign_in[:service_name]` is set upon authentication to the currently used credential.

We do have a large number of users that have multiple credential logins; in addition there are long-term efforts to move users off of DSLogon and MHV to Login.gov & ID.me so accounting for users with multiple verified credentials should be done. My suggestion would be to use the `UserAccount` to look up all of its child `UserVerification` objects so that you can check each of their `verified_at` timestamps.

```ruby
user_verifications = UserVerification.where(user_account_id: user.user_account_uuid)
user_verifications.each do |user_verification|
            days_since_verification = (Time.zone.today - user_verification.verified_at.to_date).to_i
end
```

Reply:
Ah of course.  Yes that makes sense.  I updated my code -- appreciate it!